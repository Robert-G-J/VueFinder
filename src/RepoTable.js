import React from "react";
import "./RepoTable.css";

class RepoTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      repos: []
    };
  }
  fetchRepos() {
    const that = this;
    fetch(
      "https://api.github.com/search/repositories?q=vue+language:javascript&sort=stars&order=desc"
    )
      .then(res => res.json())
      .then(res => that.setState({ repos: [...res.items] }));
  }

  componentDidMount() {
    this.fetchRepos();
  }
  render() {
    return (
      <table>
        <thead>
          <tr>
            <th>Name</th>
            <th>Description</th>
            <th>Stars</th>
          </tr>
        </thead>
        <tbody>
          {this.state.repos.map(repo => (
            <tr key={repo.id}>
              <td>{repo.full_name}</td>
              <td>{repo.description}</td>
              <td>{repo.stargazers_count}</td>
            </tr>
          ))}
        </tbody>
      </table>
    );
  }
}

export default RepoTable;
