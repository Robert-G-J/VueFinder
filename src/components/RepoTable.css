/* Styles have been normalized with normalize.css */

* {
  box-sizing: border-box;
}

body {
  font-family: Roboto, Helvetica, sans-serif;
}

/* Table Styles */

table {
  width: 100%;
}

td {
  padding-right: 0.5rem;
}

th {
  border-bottom: 1px solid #eee;
}

.repo-table th,
.repo-table td {
  text-align: left;
}

td.description {
  font-weight: 300;
}

td.stars,
th.stars {
  text-align: right;
  padding-right: 0.5rem;
}

th.header {
  padding-bottom: 0.5rem;
}

td.row {
  padding-top: 0.25rem;
  padding-bottom: 0.25rem;
}

.bp-tablet {
  display: none;
}

@media (min-width: 768px) {
  .bp-tablet {
    display: block;
  }
}
