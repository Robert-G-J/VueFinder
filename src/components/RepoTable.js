import React from "react";
import fetchFromGithub from "../services/github";
import "./RepoTable.css";

class RepoTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      repos: []
    };
  }
  fetchRepos = async () => {
    try {
      const repos = await fetchFromGithub();
      this.setState({
        repos: [...repos]
      });
    } catch (error) {
      console.error(error);
    }
  };

  componentDidMount() {
    this.fetchRepos();
  }
  render() {
    return (
      <table className="repo-table">
        <caption>Most Starred Vue Repos</caption>
        <thead className="table-head">
          <tr>
            <th className="header">Name</th>
            <th className="header bp-tablet">Description</th>
            <th className="header stars">Stars</th>
          </tr>
        </thead>
        <tbody>
          {this.state.repos.map(repo => (
            <tr key={repo.id}>
              <td className="row name">{repo.full_name}</td>
              <td className="row description bp-tablet">{repo.description}</td>
              <td className="row stars">{repo.stargazers_count}</td>
            </tr>
          ))}
        </tbody>
      </table>
    );
  }
}

export default RepoTable;
