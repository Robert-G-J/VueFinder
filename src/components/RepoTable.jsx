import React from "react";
import fetchFromGithub from "../services/github";
import "./RepoTable.css";

class RepoTable extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      repos: []
    };
  }

  async componentDidMount() {
    try {
      const repos = await fetchFromGithub();
      this.setState({
        repos: [...repos]
      });
    } catch (error) {
      console.error("Request failed");
    }
  }

  render() {
    const { repos } = this.state;
    return (
      <table className="repo-table">
        <caption>Most Starred Vue Repos</caption>
        <thead className="table-head">
          <tr>
            <th className="header">Name</th>
            <th className="header bp-tablet">Description</th>
            <th className="header stars">Stars</th>
          </tr>
        </thead>
        <tbody>
          {repos.map(repo => (
            <tr key={repo.id}>
              <td className="row name">{repo.full_name}</td>
              <td className="row description bp-tablet">{repo.description}</td>
              <td className="row stars">{repo.stargazers_count}</td>
            </tr>
          ))}
        </tbody>
      </table>
    );
  }
}

export default RepoTable;
