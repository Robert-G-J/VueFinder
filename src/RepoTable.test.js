import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import RepoTable from "./RepoTable";

describe("<RepoTable/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<RepoTable />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const component = shallow(<RepoTable />);
    expect(component.exists()).toEqual(true);
  });
});
