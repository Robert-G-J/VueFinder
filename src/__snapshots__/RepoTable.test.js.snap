// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoTable/> renders correctly 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Name
      </th>
      <th>
        Description
      </th>
      <th>
        Stars
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="name"
      />
      <td
        className="description"
      />
      <td
        className="stars"
      />
    </tr>
  </tbody>
</table>
`;
